`TBB` link interface causing error on Windows

Description
---

It looks like I broke something in a previous PR exporting a link interface to `TBB` (https://github.com/xtensor-stack/xtensor/pull/2456).
On Windows, trying to build again a project with `XTENSOR_USE_TBB` gives me the following output:
```
(libthermo) (...) >ninja
[1/1] Linking CXX shared module pythermo\pythermo_core.cp311-win_amd64.pyd
FAILED: pythermo/pythermo_core.cp311-win_amd64.pyd
cmd.exe /C "cd . && C:\Users\adrie\micromamba\envs\libthermo\Library\bin\cmake.exe -E vs_link_dll --intdir=pythermo\CMakeFiles\pythermo_core.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~2\2019\BUILDT~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe /nologo pythermo\CMakeFiles\pythermo_core.dir\src\main.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\thermo.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\ideal_gas.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\poly_gas.cpp.obj  /out:pythermo\pythermo_core.cp311-win_amd64.pyd /implib:pythermo\pythermo_core.lib /pdb:pythermo\pythermo_core.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO  -LTCG  C:\Users\adrie\micromamba\envs\libthermo\Library\lib\tbb12.lib  C:\Users\adrie\micromamba\envs\libthermo\libs\python311.lib  tbb.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK: command "C:\PROGRA~2\MICROS~2\2019\BUILDT~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe /nologo pythermo\CMakeFiles\pythermo_core.dir\src\main.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\thermo.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\ideal_gas.cpp.obj pythermo\CMakeFiles\pythermo_core.dir\src\poly_gas.cpp.obj /out:pythermo\pythermo_core.cp311-win_amd64.pyd /implib:pythermo\pythermo_core.lib /pdb:pythermo\pythermo_core.pdb /dll /version:0.0 /machine:x64 /INCREMENTAL:NO -LTCG C:\Users\adrie\micromamba\envs\libthermo\Library\lib\tbb12.lib C:\Users\adrie\micromamba\envs\libthermo\libs\python311.lib tbb.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:pythermo\pythermo_core.cp311-win_amd64.pyd.manifest" failed (exit code 1181) with the following output:
LINK : fatal error LNK1181: cannot open input file 'tbb.lib'
ninja: build stopped: subcommand failed.
```

`C:\Users\adrie\micromamba\envs\libthermo\Library\lib\tbb12.lib` is added by my downstream project using `TBB::tbb` target while `tbb.lib` comes from `xtensor`. It also fails without linking against `TBB::tbb` target.
It looks like `TBB::tbb` is the [correct way](https://github.com/oneapi-src/oneTBB/blob/9e15720bc7744f85dff611d34d65e9099e077da4/examples/common/cmake/modules/FindTBB.cmake#L47) to link against `TBB`.

I just don't really understand why it started to fail recently. It also appears that Linux builds are not affected.
Any clue about that?

--- a/xtensorConfig.cmake.in
+++ b/xtensorConfig.cmake.in
@@ -32,7 +32,7 @@ endif()
 
 if(XTENSOR_USE_TBB)
     find_dependency(TBB)
-    target_link_libraries(@PROJECT_NAME@ INTERFACE tbb)
+    target_link_libraries(@PROJECT_NAME@ INTERFACE TBB::tbb)
     target_compile_definitions(@PROJECT_NAME@ INTERFACE XTENSOR_USE_TBB)
 endif()
 

